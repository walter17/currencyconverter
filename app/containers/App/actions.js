/*
 * App Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  LOAD_RATELIST,
  LOAD_RATELIST_SUCCESS,
  LOAD_RATELIST_ERROR,
  CONVERT,
  CONVERT_SUCCESS,
  CONVERT_ERROR,
} from './constants';

/**
 * Load current currency exchange rates, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_RATELIST
 */
export function loadExchangeList() {
  return {
    type: LOAD_RATELIST,
  };
}

/**
 * Dispatched when the retrieving data succeeds
 *
 * @param  {array} list The exchange rates data
 *
 * @return {object} An action object with a type of LOAD_RATELIST_SUCCESS passing the list
 */
export function exchangeListSuccess(list) {
  return {
    type: LOAD_RATELIST_SUCCESS,
    list,
  };
}

/**
 * Dispatched when retrieving the list fails
 *
 * @param  {object} error The error
 *
 * @return {object} An action object with a type of LOAD_RATELIST_ERROR passing the error
 */
export function exchangeListError(error) {
  return {
    type: LOAD_RATELIST_ERROR,
    error,
  };
}

/**
 * Triggered to process currency conversion, this action starts the request saga
 *
 * @return {object} An action object with a type of CONVERT
 */
export function sendToConvert(leftCurrency, rightCurrency, valueToConvert) {
  return {
    type: CONVERT,
    leftCurrency,
    rightCurrency,
    valueToConvert,
  };
}

/**
 * Dispatched when the convert process succeeds
 *
 * @return {object} An action object with a type of CONVERT_SUCCESS
 */
export function convertSuccess(info) {
  return {
    type: CONVERT_SUCCESS,
    info,
  };
}

/**
 * Dispatched when currency conversion fails
 *
 * @param  {object} error The error
 *
 * @return {object} An action object with a type of CONVERT_ERROR passing the error
 */
export function convertFailed(error) {
  return {
    type: CONVERT_ERROR,
    error,
  };
}
