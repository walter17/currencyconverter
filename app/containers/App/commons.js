/**
 * Common Functions Handlers | Globally accessed functions
 * -------------------------------------------------------
 * Author: Walter Ramos
 * Last Modified: 03/01/21
 */

// Currency names arr, that is not part of the api response values
export function getCountryByCode(code) {
  const currNamesArr = [
    {
      code: 'AED',
      name: 'United Arab Emirates Dirham',
    },
    { code: 'AFN', name: 'Afghan afghani' },
    { code: 'ALL', name: 'Albania Lek' },
    { code: 'AMD', name: 'Armenian dram' },
    { code: 'ANG', name: 'Netherlands Antillean guilder' },
    { code: 'AOA', name: 'Angolan kwanza' },
    { code: 'ARS', name: 'Argentine peso' },
    { code: 'AUD', name: 'Australian dollar' },
    { code: 'AWG', name: 'Aruban florin' },
    { code: 'AZN', name: 'Azerbaijani manat' },
    { code: 'BAM', name: 'Bosnia-Herzegovina Convertible Marka' },
    { code: 'BBD', name: 'Barbados Dollar' },
    { code: 'BDT', name: 'Bangladeshi taka' },
    { code: 'BGN', name: 'Bulgarian lev' },
    { code: 'BHD', name: 'Bahraini dinar' },
    { code: 'BIF', name: 'Burundian franc' },
    { code: 'BMD', name: 'Bermuda Dollar' },
    { code: 'BND', name: 'Brunei dollar' },
    { code: 'BOB', name: 'Bolivian Boliviano' },
    { code: 'BRL', name: 'Brazilian real' },
    { code: 'BSD', name: 'Bahamian dollar' },
    { code: 'BTC', name: 'Bitcoin' },
    { code: 'BTN', name: 'Bhutanese ngultrum' },
    { code: 'BWP', name: 'Botswana pula' },
    { code: 'BYN', name: 'Belarusian ruble' },
    { code: 'BYR', name: 'Belarusian ruble' },
    { code: 'BZD', name: 'Belize dollar' },
    { code: 'CAD', name: 'Canadian dollar' },
    { code: 'CDF', name: 'Congolese franc' },
    { code: 'CHF', name: 'Swiss franc' },
    { code: 'CLF', name: 'Chilean Unit of Account' },
    { code: 'CLP', name: 'Chilean Peso' },
    { code: 'CNY', name: 'Chinese yuan renminbi' },
    { code: 'COP', name: 'Colombian peso' },
    { code: 'CRC', name: 'Costa Rican Colón' },
    { code: 'CUC', name: 'Cuban Convertible Peso' },
    { code: 'CUP', name: 'Cuban Peso' },
    { code: 'CVE', name: 'Cape Verdean Escudo' },
    { code: 'CZK', name: 'Czech Koruna' },
    { code: 'DJF', name: 'Djiboutian Franc' },
    { code: 'DKK', name: 'Danish krone' },
    { code: 'DOP', name: 'Dominican Peso' },
    { code: 'DZD', name: 'Algerian Dinar' },
    { code: 'EGP', name: 'Egyptian Pound' },
    { code: 'ERN', name: 'Eritrean nakfa' },
    { code: 'ETB', name: 'Ethiopian Birr' },
    { code: 'EUR', name: 'Euro' },
    { code: 'FJD', name: 'Fijian Dollar' },
    { code: 'FKP', name: 'Falkland Islands Pound' },
    { code: 'GBP', name: 'British pound sterling' },
    { code: 'GEL', name: 'Georgian lari' },
    { code: 'GGP', name: 'Guernsey pound' },
    { code: 'GHS', name: 'Ghanaian cedi' },
    { code: 'GIP', name: 'Gibraltar pound' },
    { code: 'GMD', name: 'Gambian dalasi' },
    { code: 'GNF', name: 'Guinean Franc' },
    { code: 'GTQ', name: 'Guatemalan Quetzal' },
    { code: 'GYD', name: 'Guyanaese Dollar' },
    { code: 'HKD', name: 'Hong Kong Dollar' },
    { code: 'HNL', name: 'Honduran Lempira' },
    { code: 'HRK', name: 'Croatian Kuna' },
    { code: 'HTG', name: 'Haitian Gourde' },
    { code: 'HUF', name: 'Hungarian forint' },
    { code: 'IDR', name: 'Indonesian Rupiah' },
    { code: 'ILS', name: 'Israeli New Shekel' },
    { code: 'IMP', name: 'Manx pound' },
    { code: 'INR', name: 'Indian Rupee' },
    { code: 'IQD', name: 'Iraqi dinar' },
    { code: 'IRR', name: 'Iranian Rial' },
    { code: 'ISK', name: 'Icelandic krona' },
    { code: 'JEP', name: 'Jersey pound' },
    { code: 'JMD', name: 'Jamaican Dollar' },
    { code: 'JOD', name: 'Jordanian Dinar' },
    { code: 'JPY', name: 'Japanese yen' },
    { code: 'KES', name: 'Kenyan shilling' },
    { code: 'KGS', name: 'Kyrgystani Som' },
    { code: 'KHR', name: 'Cambodian riel' },
    { code: 'KMF', name: 'Comorian franc' },
    { code: 'KPW', name: 'North Korean won' },
    { code: 'KRW', name: 'South Korean won' },
    { code: 'KWD', name: 'Kuwaiti Dinar' },
    { code: 'KYD', name: 'Cayman Islands Dollar' },
    { code: 'KZT', name: 'Kazakhstani Tenge' },
    { code: 'LAK', name: 'Laotian Kip' },
    { code: 'LBP', name: 'Lebanese pound' },
    { code: 'LKR', name: 'Sri Lankan Rupee' },
    { code: 'LRD', name: 'Liberian Dollar' },
    { code: 'LSL', name: 'Lesotho loti' },
    { code: 'LTL', name: 'Lithuanian litas' },
    { code: 'LVL', name: 'Latvian lats' },
    { code: 'LYD', name: 'Libyan Dinar' },
    { code: 'MAD', name: 'Moroccan Dirham' },
    { code: 'MDL', name: 'Moldovan Leu' },
    { code: 'MGA', name: 'Malagasy Ariary' },
    { code: 'MKD', name: 'Macedonian Denar' },
    { code: 'MMK', name: 'Myanmar kyat' },
    { code: 'MNT', name: 'Mongolian tögrög' },
    { code: 'MOP', name: 'Macau pataca' },
    { code: 'MRO', name: 'Mauritanian ouguiya' },
    { code: 'MUR', name: 'Mauritian rupee' },
    { code: 'MVR', name: 'Maldivian rufiyaa' },
    { code: 'MWK', name: 'Malawian Kwacha' },
    { code: 'MXN', name: 'Mexican Peso' },
    { code: 'MYR', name: 'Malaysian Ringgit' },
    { code: 'MZN', name: 'Mozambican metical' },
    { code: 'NAD', name: 'Namibian dollar' },
    { code: 'NGN', name: 'Nigerian naira' },
    { code: 'NIO', name: 'Nicaraguan Córdoba' },
    { code: 'NOK', name: 'Norwegian krone' },
    { code: 'NPR', name: 'Nepalese rupee' },
    { code: 'NZD', name: 'New Zealand dollar' },
    { code: 'OMR', name: 'Omani rial' },
    { code: 'PAB', name: 'Panamanian Balboa' },
    { code: 'PEN', name: 'Peruvian Sol' },
    { code: 'PGK', name: 'Papua New Guinea' },
    { code: 'PHP', name: 'Philippine peso' },
    { code: 'PKR', name: 'Pakistani rupee' },
    { code: 'PLN', name: 'Polish złoty' },
    { code: 'PYG', name: 'Paraguayan guaraní' },
    { code: 'QAR', name: 'Paraguayan guaraní' },
    { code: 'RON', name: 'Romanian leu' },
    { code: 'RSD', name: 'Serbian dinar' },
    { code: 'RUB', name: 'Russian ruble' },
    { code: 'RWF', name: 'Rwandan franc' },
    { code: 'SAR', name: 'Saudi Riyal' },
    { code: 'SBD', name: 'Solomon Islands Dollar' },
    { code: 'SCR', name: 'Seychellois Rupee' },
    { code: 'SDG', name: 'Sudanese pound' },
    { code: 'SEK', name: 'Swedish Krona' },
    { code: 'SGD', name: 'Singapore dollar' },
    { code: 'SHP', name: 'Saint Helena pound' },
    { code: 'SLL', name: 'Sierra Leonean Leone' },
    { code: 'SOS', name: 'Somali Shilling' },
    { code: 'SRD', name: 'Surinamese Dollar' },
    { code: 'STD', name: 'São Tomé and Príncipe dobra' },
    { code: 'SVC', name: 'Salvadoran colón' },
    { code: 'SYP', name: 'Syrian pound' },
    { code: 'SZL', name: 'Swazi Lilangeni' },
    { code: 'THB', name: 'Thai baht' },
    { code: 'TJS', name: 'Tajikistani Somoni' },
    { code: 'TMT', name: 'Turkmenistan manat' },
    { code: 'TND', name: 'Tunisian dinar' },
    { code: 'TOP', name: "Tonga Pa'anga" },
    { code: 'TRY', name: 'Turkish lira' },
    { code: 'TTD', name: 'Trinidad and Tobago dollar' },
    { code: 'TWD', name: 'New Taiwan dollar' },
    { code: 'TZS', name: 'Tanzanian Shilling' },
    { code: 'UAH', name: 'Ukrainian hryvnia' },
    { code: 'UGX', name: 'Ugandan Shilling' },
    { code: 'USD', name: 'United States Dollar' },
    { code: 'UYU', name: 'Uruguayan Peso' },
    { code: 'UZS', name: 'Uzbekistani Som' },
    { code: 'VEF', name: 'Venezuelan bolivar' },
    { code: 'VND', name: 'Vietnamese dong' },
    { code: 'VUV', name: 'Vanuatu vatu' },
    { code: 'WST', name: 'Samoan tālā' },
    { code: 'XAF', name: 'Central African CFA franc' },
    { code: 'XAG', name: 'Silver Ounce' },
    { code: 'XAU', name: 'Gold' },
    { code: 'XCD', name: 'Eastern Caribbean dollar' },
    { code: 'XDR', name: 'Special Drawing Rights' },
    { code: 'XOF', name: 'West African CFA franc' },
    { code: 'XPF', name: 'CFP franc' },
    { code: 'YER', name: 'Yemeni Rial' },
    { code: 'ZAR', name: 'South African Rand' },
    { code: 'ZMK', name: 'Zambian Kwacha' },
    { code: 'ZMW', name: 'Zambian kwacha' },
    { code: 'ZWL', name: 'Zimbabwean dollar' },
  ];
  return currNamesArr.filter(data => data.code === code);
}
