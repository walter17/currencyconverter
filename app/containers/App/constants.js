/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

// Constants --- Retrieve Exhange Rate List
export const LOAD_RATELIST = 'containers/ExchangeRates/LOAD_RATELIST';
export const LOAD_RATELIST_SUCCESS =
  'containers/ExchangeRates/LOAD_RATELIST_SUCCESS';
export const LOAD_RATELIST_ERROR =
  'containers/ExchangeRates/LOAD_RATELIST_ERROR';

// API Constants
const API_BASE_URL = `http://data.fixer.io/api/`;
export const CURRENCY_LIST_URI = `${API_BASE_URL}latest`; // API to GET latest currency rate list
export const CONVERT_URI = `${API_BASE_URL}convert`; // URI to process currency conversions
