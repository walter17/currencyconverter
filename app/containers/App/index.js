/**
 *
 * App
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 */

import React from 'react';
import { Helmet } from 'react-helmet';
import styled from 'styled-components';
import { Switch, Route } from 'react-router-dom';

import NotFoundPage from 'containers/NotFoundPage/Loadable';
import Header from 'components/Header';

import GlobalStyle from '../../global-styles';

const AppWrapper = styled.div`
  max-width: calc(768px + 16px * 2);
  margin: 0 auto;
  display: flex;
  min-height: 100%;
  padding: 0 16px;
  flex-direction: column;
`;

export default function App() {
  return (
    <AppWrapper>
      <Helmet
        titleTemplate="%s - Currency Converter"
        defaultTitle="Currency Converter"
      >
        <meta name="description" content="A Currency Converter application" />
      </Helmet>
      <Header />
      <Switch>
        {/* Return default DIV */}
        <Route exact path="/" component={() => <div />} />
        {/* Return 404 Page if route not registered */}
        <Route path="" component={NotFoundPage} />
      </Switch>
      <GlobalStyle />
    </AppWrapper>
  );
}
