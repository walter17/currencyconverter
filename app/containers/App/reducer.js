/*
 * AppReducer
 *
 * The reducer takes care of our data. Using actions, we can
 * update our application state. To add a new action,
 * add it to the switch statement in the reducer function
 *
 */

import produce from 'immer';
import {
  LOAD_RATELIST,
  LOAD_RATELIST_SUCCESS,
  LOAD_RATELIST_ERROR,
  CONVERT,
  CONVERT_SUCCESS,
  CONVERT_ERROR,
} from './constants';

// The initial state of the App
export const initialState = {
  loading: false,
  error: false,
  exchangeRateList: [],
  convert: {
    loading: false,
    error: false,
    data: false,
    leftCurrency: 0,
    rightCurrency: 0,
    valueToConvert: 0,
  },
};

/* eslint-disable default-case, no-param-reassign */
const appReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case LOAD_RATELIST:
        draft.loading = true;
        draft.error = false;
        draft.exchangeRateList = [];
        break;

      case LOAD_RATELIST_SUCCESS:
        draft.exchangeRateList = action.list;
        draft.loading = false;
        draft.error = false;
        break;

      case LOAD_RATELIST_ERROR:
        draft.error = action.error;
        draft.loading = false;
        draft.exchangeRateList = [];
        break;

      case CONVERT:
        draft.convert.loading = true;
        draft.convert.error = false;
        draft.convert.data = false;
        draft.convert.leftCurrency = action.leftCurrency;
        draft.convert.rightCurrency = action.rightCurrency;
        draft.convert.valueToConvert = action.valueToConvert;
        break;

      case CONVERT_SUCCESS:
        draft.convert.data = action.info;
        draft.convert.loading = false;
        draft.convert.error = false;
        break;

      case CONVERT_ERROR:
        draft.convert.error = action.error;
        draft.convert.loading = false;
        break;
    }
  });

export default appReducer;
