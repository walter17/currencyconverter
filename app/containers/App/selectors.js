/**
 * The global state selectors
 */

import { createSelector } from 'reselect';
import { initialState } from './reducer';

const selectGlobal = state => state.global || initialState;

const selectRouter = state => state.router;

export const objReduxConv = typeVal => {
  if (typeVal === 'convert') return true;

  return false;
};

const makeSelectCurrencyList = () =>
  createSelector(
    selectGlobal,
    globalState => globalState.exchangeRateList,
  );

const makeSelectLoading = () =>
  createSelector(
    selectGlobal,
    globalState => globalState.loading,
  );

const makeSelectError = () =>
  createSelector(
    selectGlobal,
    globalState => globalState.error,
  );

const makeSelectLocation = () =>
  createSelector(
    selectRouter,
    routerState => routerState.location,
  );

const makeSelectConvertLoading = itemQuery =>
  createSelector(
    selectGlobal,
    globalState =>
      objReduxConv(itemQuery) ? globalState[itemQuery].loading : false,
  );

const makeSelectConvertError = itemQuery =>
  createSelector(
    selectGlobal,
    globalState =>
      objReduxConv(itemQuery) ? globalState[itemQuery].error : false,
  );

const makeSelectConvertValue = itemQuery =>
  createSelector(
    selectGlobal,
    globalState =>
      objReduxConv(itemQuery) ? globalState[itemQuery].valueToConvert : false,
  );

const makeSelectLeftCurrency = itemQuery =>
  createSelector(
    selectGlobal,
    globalState =>
      objReduxConv(itemQuery) ? globalState[itemQuery].leftCurrency : false,
  );

const makeSelectRightCurrency = itemQuery =>
  createSelector(
    selectGlobal,
    globalState =>
      objReduxConv(itemQuery) ? globalState[itemQuery].rightCurrency : false,
  );

const makeSelectConvertData = itemQuery =>
  createSelector(
    selectGlobal,
    globalState =>
      objReduxConv(itemQuery) ? globalState[itemQuery].data : false,
  );

export {
  selectGlobal,
  makeSelectCurrencyList,
  makeSelectLoading,
  makeSelectError,
  makeSelectLocation,
  makeSelectConvertLoading,
  makeSelectConvertError,
  makeSelectConvertValue,
  makeSelectLeftCurrency,
  makeSelectRightCurrency,
  makeSelectConvertData,
};
