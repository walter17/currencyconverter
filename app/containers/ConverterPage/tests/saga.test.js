/**
 * Currency Converter Page Unit Testing Saga Coverage
 */
import { put, takeLatest } from 'redux-saga/effects';

import { LOAD_RATELIST, CONVERT } from 'containers/App/constants';
import {
  exchangeListSuccess,
  exchangeListError,
  convertSuccess,
  convertFailed,
} from 'containers/App/actions';

import currencyListData, { getCurrencyList, convertCurrency } from '../saga';

describe('Retrieve currency exchange rates Saga', () => {
  let getGenerator;

  beforeEach(() => {
    getGenerator = currencyListData();

    const selectDescriptor = getGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the exchangeListSuccess action when rates list retrieved successfully', () => {
    const response = [{ success: true }];
    expect({
      '@@redux-saga/IO': true,
      combinator: false,
      payload: {
        action: {
          list: response,
          type: 'containers/App/LOAD_RATELIST_SUCCESS',
        },
        channel: undefined,
      },
      type: 'PUT',
    }).toEqual(put(exchangeListSuccess(response)));
  });

  it('should call the exchangeListError action rates retreival fails', () => {
    const response = new Error('Some error');

    expect({
      '@@redux-saga/IO': true,
      combinator: false,
      payload: {
        action: {
          error: new Error('Some error'),
          type: 'containers/App/LOAD_RATELIST_ERROR',
        },
        channel: undefined,
      },
      type: 'PUT',
    }).toEqual(put(exchangeListError(response)));
  });

  it('should dispatch the convertSuccess action when converted successfully', () => {
    const response = [{ success: true }];
    expect({
      '@@redux-saga/IO': true,
      combinator: false,
      payload: {
        action: {
          info: response,
          type: 'containers/App/CONVERT_SUCCESS',
        },
        channel: undefined,
      },
      type: 'PUT',
    }).toEqual(put(convertSuccess(response)));
  });

  it('should call the convertFailed action if device login fails', () => {
    const response = new Error('Some error');

    expect({
      '@@redux-saga/IO': true,
      combinator: false,
      payload: {
        action: {
          error: new Error('Some error'),
          type: 'containers/App/CONVERT_ERROR',
        },
        channel: undefined,
      },
      type: 'PUT',
    }).toEqual(put(convertFailed(response)));
  });

  describe('currencyListData Saga', () => {
    const currencySaga = currencyListData();

    it('should start task to watch for LOAD_RATELIST action', () => {
      const takeLatestDescriptor = currencySaga.next().value;
      expect(takeLatestDescriptor).toEqual(
        takeLatest(LOAD_RATELIST, getCurrencyList),
      );
    });

    it('should start task to watch for CONVERT action', () => {
      const takeLatestDescriptor = currencySaga.next().value;
      expect(takeLatestDescriptor).toEqual(
        takeLatest(CONVERT, convertCurrency),
      );
    });
  });
});
