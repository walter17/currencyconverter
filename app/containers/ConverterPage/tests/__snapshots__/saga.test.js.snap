// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Retrieve currency exchange rates Saga currencyListData Saga should start task to watch for CONVERT action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "containers/App/LOAD_RATELIST",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "FORK",
}
`;

exports[`Retrieve currency exchange rates Saga currencyListData Saga should start task to watch for LOAD_RATELIST action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "containers/App/LOAD_RATELIST",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "FORK",
}
`;

exports[`Retrieve currency exchange rates Saga should call the convertFailed action if device login fails 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "containers/App/LOAD_RATELIST",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "FORK",
}
`;

exports[`Retrieve currency exchange rates Saga should call the exchangeListError action rates retreival fails 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "containers/App/LOAD_RATELIST",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "FORK",
}
`;

exports[`Retrieve currency exchange rates Saga should dispatch the convertSuccess action when converted successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "containers/App/LOAD_RATELIST",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "FORK",
}
`;

exports[`Retrieve currency exchange rates Saga should dispatch the exchangeListSuccess action when rates list retrieved successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "containers/App/LOAD_RATELIST",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "FORK",
}
`;
